libres/lib/include/ert/analysis/ies/ies.hpp

```diff
@@ -39,8 +39,7 @@ void init_update(void *arg, const bool_vector_type *ens_mask,
                  const matrix_type *R, const matrix_type *dObs,
                  const matrix_type *E, const matrix_type *D, rng_type *rng);
 
-void initX(const std::variant<double, int> &truncation,
-           config::inversion_type ies_inversion, const matrix_type *Y0,
+void initX(const config::config_type *ies_config, const matrix_type *S,
```

Comment:
What do you think of adding a docstring to this in Doxygen format, something like:

```cpp
/**
 * @brief 
 * 
 * @param ies_config 
 * @param S 
 * @param R 
 * @param E 
 * @param D 
 * @param X 
 */
```

Reply:
Personally I think Doxygen means "old and irrelevant" - so I would prefer not to.

Reply:
What format of docstrings do you prefer?

Reply:
My general attitude is that I am skeptical to documentation strings, because I feel they very often are stale and irrelevant[1]. The formatted strings (typically doxygen) even more so, because I suspect the guy inserting that comment used a shortcut in his editor with thinking much about it - i.e. it screams: "Comment without love".

My preferred way to handle comments is that when something is *really* complex/broken/.... and I see no obvious way to improve the problem away I write a comment. The comment can be quite extensive, it is in free form and typically for developers actually working with the problematic code - i.e. in the .cpp file. In addition to the text in the comment the mere fact that I have actually added a comment should signal "here be dragons" for the pour soul coming after me.

But I am pushing 50 and have no spine left - will do as encouraged.


[1]: Due to `pydoc` I do feel quite different about this in Python - here the comments/document strings are available to a much larger audience and it feels meaningful to write them.

Reply:
> My general attitude is that I am skeptical to documentation strings, because I feel they very often are stale and irrelevant[1]. The formatted strings (typically doxygen) even more so, because I suspect the guy inserting that comment used a shortcut in his editor with thinking much about it - i.e. it screams: "Comment without love".

The staleness is in my opinion due to docstrings containing too much detail.
In this specific case I would write that `initX` is equivalent to a standard EnKF implementation but that it uses an iterative ensemble smoother under the hood or something like that.

> My preferred way to handle comments is that when something is _really_ complex/broken/.... and I see no obvious way to improve the problem away I write a comment. The comment can be quite extensive, it is in free form and typically for developers actually working with the problematic code - i.e. in the .cpp file. In addition to the text in the comment the mere fact that I have actually added a comment should signal "here be dragons" for the pour soul coming after me.

I think this is fine, but should ideally be dealt with fast.
In other words, such comments should not grow old, which they often do.

> But I am pushing 50 and have no spine left - will do as encouraged.

The spine hardens with age!

Reply:
I have added a comment which I am happy with.